### ISSUE TITLE
Add future and past providers to datetime

### ISSUE THREAD

**Issue by sobolevn** [lk-geimfari/mimesis#775]
I would love to see the same feature from `Datetime` ported to both `FutureDatetime` and `PastDatetime` providers.

I hope that you will get the idea from the class names 🙂

**Comment by sobolevn** [comment]
Related: https://faker.readthedocs.io/en/master/providers/faker.providers.date_time.html

**Comment by lk-geimfari** [comment]
@ceccoemi Any chance you can work on it? I'm about to publish a new version and this feature will be a great addition in the changelog.

**Comment by ceccoemi** [comment]
@lk-geimfari yes, I should be able to work on it. Give me 2/3 days

**Comment by lk-geimfari** [comment]
@ceccoemi Great! Thank you very much! Feel free to ask your questions about the feature to @sobolevn.

**Comment by ceccoemi** [comment]
@sobolevn @lk-geimfari a couple of questions.

What do you think about adding the `FutureDatetime` and `PastDatetime` providers as subclass of the `Datetime` provider? In that case, is it ok if I include them in the same module `date.py`?

My idea is to override the methods that accepts the argument `start` and to set it to a moment in the future/past.

**Comment by sobolevn** [comment]
Sounds like a plan!

**Comment by sobolevn** [comment]
This was never implemented. I needed this today once again.

**Comment by lk-geimfari** [comment]
Almost every `Datetime` method accepts ranges of years as I can see.

**Comment by sobolevn** [comment]
not in an easy way, judging from the docs here: https://mimesis.name/master/api.html#datetime

Faker has `future_datetime()` and `past_datetime` that have the semantics that I want.
But, mimesis has no such feature. I have to manually add years in the past to get such days. Moreover, year sometimes is not enough. I want the datetime in the past this year. Or this month. Or this week, etc.

There are lots of business rules based on time. These things are 100% required. See how many of them exist here: https://faker.readthedocs.io/en/stable/providers/faker.providers.date_time.html

**Comment by lk-geimfari** [comment]
> not in an easy way, judging from the docs here: https://mimesis.name/master/api.html#datetime
> 
> Faker has `future_datetime()` and `past_datetime` that have the semantics that I want. But, mimesis has no such feature. I have to manually add years in the past to get such days. Moreover, year sometimes is not enough. I want the datetime in the past this year. Or this month. Or this week, etc.
> 
> There are lots of business rules based on time. These things are 100% required. See how many of them exist here: https://faker.readthedocs.io/en/stable/providers/faker.providers.date_time.html

Can't we just add extra argument and implement something like this:

```
dt = Datetime(moment='PAST')
dt = Datetime(moment='FUTURE')
```

**Comment by sobolevn** [comment]
Yes, this looks good to me. I would suggest to add more, like: `PAST_YEAR`, `PAST_MONTH`, `PAST_WEEK`, `PAST_DAY`. And the same for `FUTURE`

**Comment by lk-geimfari** [comment]
@sobolevn Hey! I'm planning on providing something similar:

```python
class Timeframe:
    PAST_YEAR = 'PAST_YEAR'
    PAST_MONTH = 'PAST_MONTH'
    PAST_WEEK = 'PAST_WEEK'
    PAST_DAY = 'PAST_DAY'
    FUTURE_YEAR = 'FUTURE_YEAR'
    FUTURE_MONTH = 'FUTURE_MONTH'
    FUTURE_WEEK = 'FUTURE_WEEK'
    FUTURE_DAY = 'FUTURE_DAY'
    ANY = 'ANY'
    

# Note: There are only 9 method of "Datetime" to which this feature is applicable:
dt = Datetime(timeframe=Timeframe.PAST_YEAR)
dt.bulk_create_datetimes()
dt.week_date()
dt.year()
dt.date()
dt.formatted_date()
dt.time()
dt.datetime()
dt.formatted_datetime()
dt.timestamp()
```

Depending on the `Timeframe` passed in, it will always generate dates for the given time. The default will be `Timeframe.ANY`.

I can implement this on the weekend if it suits your needs.

**Comment by sobolevn** [comment]
I also propose to add: `FUTURE_THIS_DAY` and `PAST_THIS_DAY`.
Other than that - looks good to me!

**Comment by lk-geimfari** [comment]
Turns out it's not as easy as I thought it would be. Nevertheless, I still plan to implement this feature, but couldn't give an accurate estimate as I'm very busy with work.

**Comment by rodrigobnogueira** [comment]
I proposed a PR following a simple approach, pretty much as it is done in Faker.
